src/lib.rs

```diff
@@ -499,14 +499,23 @@ pub struct PatternError {
     /// The approximate character index of where the error occurred.
     pub pos: usize,
 
-    /// A message describing the error.
-    pub msg: &'static str,
+    /// Specific kind of pattern error.
+    pub msg: PatternErrorKind,
 }
 
-impl Error for PatternError {
```

Comment:
This impl should be kept, but it can just become an empty `{}` since `description` is deprecated anyway